---
output:
  md_document:
    variant: markdown_github
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "README-",
  warning = FALSE, error = FALSE
)
```
   
   
# miRTarget : Integrative analysis of miRNA and mRNA
       
     
miRTarget helps to find miRNA predicted and validated targets.  Two databases are used [miRecords](http://c1.accurascience.com/miRecords/) and [miRTarBase](http://mirtarbase.mbc.nctu.edu.tw)
     
     
     
## Installation and loading
    

   
-  Install the latest version from [GitHub](https://github.com/kassambara/miRTarget) as follow: 
    
   
```{r, eval = FALSE}
# Install
if(!require(devtools)) install.packages("devtools")
devtools::install_github("kassambara/miRTarget")
```
     
     
- Loading
   
```{r}
library(miRTarget)
```

    
    
## Examples of miRNAs
     
```{r}
miRNAs <- c("hsa-miR-28-5p", "hsa-miR-768-5p")
```
    
    
## Find miRNA targets

    
     
### Get predicted miRNA targets
   
[miRecords.v4](c1.accurascience.com/miRecords/) database is used. For each miRNA, we want to identify the predicted target genes by at least 5/11 databases.
   
The 11 databases are : "diana", "microinspector",	"miranda",	"mirtarget2", "mitarget", "nbmirtar",	"pictar", "pita",	"rna22", "rnahybrid", "targetscan".  
   
    
Internet connexion is required. 
     
     
```{r}
p_targets <- get_mirecords_pt(miRNAs, n = 5, species = "Homo sapiens",
                                      result_file = "mirna_mirecords_predicted_targets.txt",
                              show_progress = FALSE)
head(p_targets, 10)
```
    
    
### Get miRNA validated targets
   
   
miRecords and miRtarbase are used. Internet connexion is required.
    
    
1. miRecords
    
    
```{r}
mirecords_vt <- get_mirecords_vt(miRNAs, 
                                 species = "Homo sapiens", version = 4)
head(mirecords_vt, 10)
```
    
    
2. miRtarbase
     
     
```{r}
mirtarbase_vt <- get_mirtarbase_vt(miRNAs)
head(mirtarbase_vt, 10)
```
    
    
3. Combine validated targets
    
    
```{r}
v_targets <- combine_mir_vt(mirecords_vt, mirtarbase_vt)
# head(v_targets) 
```
    
    
### Merge predicted and validated targets   
     
     
```{r}
targets <- combine_mir_pvt(p_targets, v_targets)
head(targets)
```
    
    
## Correlation between miRNA expression and target gene expressions
    
    
### Expression values of miRNAs

```{r}
# Create some data with 4 samples (S) and two miRNAs
mirna_exprs <- data.frame(S1 = c(99, 8690), S2 = c(31, 612),
                          S3 = c(33, 1747), S4 = c(29, 2540))
rownames(mirna_exprs) <- c("hsa-miR-28-5p", "hsa-miR-768-5p")

print(mirna_exprs)
```
    
    
### Expression value of miRNA potential target genes
     
     
```{r}
data("gene_exprs")
head(gene_exprs, 10)
```
     
     

### Keep only targets, which gene expressions are available
   
   
```{r}
# Keep only target with gene expression data
targets <- subset(targets, targets$target.gene_name %in% rownames(gene_exprs) )
# Order data by miRNA name
targets <- targets[order(targets$mirna_name),  ]
# Change rownames
rownames(targets) <- paste0(targets$mirna_name, "::", targets$target.gene_name)
head(targets)
```

    
### Combine miRNA and mRNA expression data
      
      
The name and the number of samples in the two data set should be identical.
      
```{r}
mir_mrna_exprs <- combine_mir_mrna_exprs(mirna_exprs, gene_exprs)
```
      
      
### Correlation between miRNAs and the corresponding target genes
      
      
> If a gene is targeted by a miRNA we expect that the expression profile of the gene and the miRNA are anti-correlated.
     
     
```{r}
res.cor <- corr_mir_mrna( mir_targets = MirTarget(targets),
                          mir_mrna_exprs = mir_mrna_exprs, 
                          show_progress = FALSE, stand = TRUE)
```
     
     
     
- Number of mRNAs per miRNAs with an absolute correlation of 0.6
   
   
```{r, fig.width = 4, fig.height=4}
viz_mrna_per_mir(res.cor, coeff = 0.6)
```
    
    
- If you want to visualize the fraction of negative correlation use this
    
```{r}
viz_neg_cor_fraction(res.cor)
```
     
     

    
### Keep only negative correlation
    
    
r < -0.6
    
     
```{r}
# Negative correlation
neg.cor <- subset(res.cor, abs(cor.coeff) > 0.6 & cor.coeff < 0)
head(neg.cor, 10)

nmir <- length(unique(neg.cor$mirna_name))
ngn <- length(unique(neg.cor$target.gene_name))
valid <- subset(neg.cor, status %in% c("PV", "V"))
```
      
      
    
- Number of negative interactions: `r nrow(neg.cor)`
- Number of unique gene : `r ngn`
- Number of unique miRNA : `r nmir`
- Validated target-interaction: `r nrow(valid)`
    - Unique genes : `r length(unique(valid$target.gene_name))`
    - Unique miRNAs : `r length(unique(valid$mirna_name))`
   
   
     
